internal/npm_install/generate_build_file.ts

```diff
@@ -76,42 +75,58 @@ package(default_visibility = ["${visibility}"])
 }
 
 if (require.main === module) {
-  main();
+  main()
+}
+
+
+async function exists(p: string) {
+  return await fs.access(p, constants.F_OK | constants.W_OK).then(() => true, () => false);
 }
 
+// Avoid duplicate mkdir call when mkdir runs parallel.
```

Comment:
The `exists` avoids the `mkdir`, this avoids creating a (possibly unnecessary) promise to do that `exists`. Is creating this potentially giant `Map` worth avoid that extra promise and `exists` check?

Reply:
"mkdirp" now runs in parallel and no longer has a strict execution order.
for example
```js
// async version
await Promise.all([mkdirp('@test/foo') , mkdirp('@test/bar')])

/*
  call1 exists('@test/foo') -> false
  call2 exists('@test/bar') -> false
  call1 exists('@test') -> false
  call2 exists('@test') -> false
  call1 fs.mkdir('@test') -> pass
  call2 fs.mkdir('@test') -> error
*/

// sync version with fs.existsSync and fs.mkdirSync
mkdirp('@test/foo')
mkdirp('@test/bar')
/*
  call1 exists('@test/foo') -> false
  call1 exists('@test') -> false
  call1 fs.mkdir('@test') -> pass
  call1 fs.mkdir('@test/foo') -> pass

  call2 exists('@test/bar') -> false
  call2 exists('@test') -> true
  call2 fs.mkdir('@test/bar') -> pass
*/
```